sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -119,8 +119,15 @@ case class GpuGetArrayItem(child: Expression, ordinal: Expression)
 
   override def doColumnar(lhs: GpuColumnVector, ordinal: Scalar): ColumnVector = {
     // Need to handle negative indexes...
-    if (ordinal.isValid && ordinal.getInt >= 0) {
-      lhs.getBase.extractListElement(ordinal.getInt)
+    if (ordinal.isValid) {
+      val minNumElements = lhs.getBase.countElements.min.getInt
+      if ( (ordinal.getInt < 0 || minNumElements < ordinal.getInt + 1) && failOnError) {
```

Comment:
What is supposed to happen with a `null` array?  `countElements` will return a `null` for a `null` array and min skips over nulls unless all of them are null.  So is `null[1]` an error in ansi mode or not? If it is then this code will completely miss it. If it is not an error, then we will get an exception, or possibly data corruption when we try to get the int value from the result of `min` if the batch is all null arrays.

Reply:
I did a simple test about this problem:

```
// row data is like:

+------------------------+
|col_1                   |
+------------------------+
|null                    |
|[Java, Scala, C++, a, b]|
+------------------------+

df.select(col("col_1")[2]).show()

+--------+
|col_1[2]|
+--------+
|    null|
|     C++|
+--------+
```
CPU and GPU will return the same result for this case.
But like you said, error occurs when the column contains all null arrays.  
the `countElements` works well, but the `min` will return 0 for this case and will throw the exception here.
(CPU will still return null for them)

Reply:
Updated.